apps/emqx_durable_storage/src/emqx_ds_replication_layer_meta.erl

```diff
@@ -346,7 +436,7 @@ create_shards(DB, NShards, ReplicationFactor) ->
             Hashes0 = [{hash(Shard, Site), Site} || Site <- AllSites],
             Hashes = lists:sort(Hashes0),
             {_, Sites} = lists:unzip(Hashes),
-            [First | _] = ReplicaSet = lists:sublist(Sites, 1, ReplicationFactor),
+            [First | ReplicaSet] = lists:sublist(Sites, 1, ReplicationFactor),
```

Comment:
Q: shouldn't the whole replicaset also include the first node, since it also contains the data?

Reply:
I think these two place have a little conflict https://github.com/emqx/emqx/blob/363c56c0857b076d07db678e6a78deaed8a2e9b3/apps/emqx_durable_storage/src/emqx_ds_replication_layer_meta.erl#L162, if the `ReplicaSet` included all sites, here should not check the `InSync` set?

Reply:
My understanding was that replica set means the set of all nodes that contain a replica of the data, and in-sync was the subset of those that are caught up with replication.